OloEngine/src/OloEngine/Serialization/MeshBinarySerializer.cpp

```
@@ -0,0 +1,1352 @@
+                    ReadBytes(payload, encoded.Data.data(), encoded.Data.size());
+
+                    vertices.SetNum(static_cast<i32>(geo.VertexCount));
+                    if (!MeshOptimization::DecodeVertexBuffer(vertices.GetData(), geo.VertexCount, geo.VertexStride, encoded))
+                    {
+                        OLO_CORE_ERROR("MeshBinarySerializer::Read: Failed to decode vertex buffer");
+                        return nullptr;
+                    }
+                }
+
+                // Decode indices
+                if (geo.IndexCount > 0 && geo.EncodedIndexSize > 0)
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)